bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/auth/client/oauth2/AccessTokenResponse.java

```diff
@@ -0,0 +1,223 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+
+/**
+ * This is the Access Token Response, a simple value-object that holds the result of the
+ * from an Access Token Request, as listed in RFC 6749:
+ *
+ * 4.1.4 - Authorization Code grant - Access Token Response,
+ * 4.2.2 - Implicit Grant - Access Token Response,
+ * 4.3.3 - Resource Owner Password Credentials Grant - Access Token Response
+ * 4.4.3 - Client Credentials Grant - Access Token Response
+ *
+ * @author Michael Bock - Initial contribution
+ * @author Gary Tse - Adaptation for Eclipse Smarthome
+ *
+ */
+public final class AccessTokenResponse implements Serializable, Cloneable {
+
+    /**
+     * For Serializable
+     */
+    private static final long serialVersionUID = 4837164195629364014L;
+
+    /**
+     * The access token issued by the authorization server. It is used
+     * by the client to gain access to a resource.
+     *
+     * @see https://tools.ietf.org/html/rfc6749#section-1.4
+     *
+     *      This token must be confidential in transit and storage.
+     * @see https://tools.ietf.org/html/rfc6749#section-10.3
+     */
+    private String accessToken;
+
+    /**
+     * Token type. e.g. Bearer, MAC
+     *
+     * @see https://tools.ietf.org/html/rfc6749#section-7.1
+     */
+    private String tokenType;
+
+    /**
+     * Number of seconds that this OAuthToken is valid for since the time it was created.
+     *
+     * @see https://tools.ietf.org/html/rfc6749#section-4.2.2
+     *
+     */
+    private long expiresIn;
+
+    /**
+     * Refresh token is a string representing the authorization granted to
+     * the client by the resource owner. Unlike access tokens, refresh tokens are
+     * intended for use only with authorization servers and are never sent
+     * to resource servers.
+     *
+     * @see https://tools.ietf.org/html/rfc6749#section-1.5
+     *
+     *      This token must be confidential in transit and storage.
+     * @see https://tools.ietf.org/html/rfc6749#section-10.4
+     */
+    private String refreshToken;
+
+    /**
+     * A space-delimited case-sensitive un-ordered string. This may be used
+     * by the authorization server to inform the client of the scope of the access token
+     * issued.
+     *
+     * @see https://tools.ietf.org/html/rfc6749#section-3.3
+     */
+    private String scope;
+
+    /**
+     * If the {@code state} parameter was present in the access token request.
+     * The exact value should be returned as-is from the authorization provider.
+     *
+     * https://tools.ietf.org/html/rfc6749#section-4.2.2
+     */
+    private String state;
+
+    /**
+     * Created datetime of this access token. This is generated locally
+     * by the OAUTH client as at the time the access token is received.
+     *
+     * This should be slightly later than the actual time the access token
+     * is produced at the server.
+     */
+    private Date created;
```

Comment:
I am wondering if LocalDateTime can be used instead of Date.

Reply:
Good suggestion.  Thanks